java/arcs/core/data/expression/Expression.kt

```diff
@@ -423,6 +426,30 @@ sealed class Expression<out T> {
         override fun <Result> accept(visitor: Visitor<Result>): Result = visitor.visit(this)
         override fun toString() = this.stringify()
     }
+
+    /**
+     * Represents an expression that sorts a sequence to produce a new sequence. Note that this
+     * operator can be expensive, and must traverse its qualifying sequence entirely.
+     *
+     * [selectors] is a non-empty list of expressions that return values to be used for ordering.
+     */
+    data class OrderByExpression<T>(
+        override val qualifier: Expression<Sequence<Unit>>,
+        val selectors: List<Expression<Any>>,
+        val descending: Boolean
```

Comment:
Descending should be specified per selector, e.g. see an example of LINQ here: https://stackoverflow.com/questions/298725/multiple-order-by-in-linq

I do think we should do this properly now and not leave that as a cleanup for later - it's rather fundamental even if it will not come up soon.

Alternatively - either allow orderby with a single selector only, all allow multiple selectors, but allow specifying descending per selector.

Reply:
Done